watertap/flowsheets/dye_desalination/dye_desalination.py

```diff
@@ -716,11 +716,21 @@ def add_costing(m):
             ),
             doc="Specific energy consumption of the treatment train on a feed flowrate basis [kWh/m3]",
         )
-    else:
+    elif hasattr(m.fs, "dewaterer") or hasattr(m.fs, "gac"):
+        m.fs.ro_costing.electricity_cost = value(m.fs.zo_costing.electricity_cost)
+        m.fs.ro_costing.base_currency = pyunits.USD_2020
```

Comment:
Why do these lines fall under this conditional that checks for dewatering unit or gac? Setting electricity cost and base currency seem like something that should happen so long as m.fs.ro_costing exists. Maybe there is more to the logic here that I am missing.

Reply:
Yes, you are correct